### DISCUSSION TITLE
Trino SQL validator

### DISCUSSION THREAD

**Discussion by giftig** [apache/superset#23851] [Ideas]
We've noticed that there's no trino SQL validator present, even though there's a presto one. The presto one more or less works as-is, except that the exception type retrieved from the database is different.

We've solved this in our fork of the project by extending the presto validator and switching the exception type and we'd like to contribute that change.

**Comment by giftig** [comment]
PR here: https://github.com/apache/superset/pull/23852

**Comment by rusackas** [comment]
CC a few friendly Trino folks!
@cccs-tom
@villebro
@sujiplr

**Comment by giftig** [comment]
I've closed the linked PR as it looks like what we have at the moment is a bit too much of a fudge to be more generally useful, but I'll leave the discussion open as I'd be interested in input on what we want a "better" trino SQL validator to do. Our team have some of our own ideas on that, too, and we already have some plans to add some quite customised warnings / errors on top of whatever we end up with here, e.g. to warn users that the query they're writing has a lot of joins / will cover a lot of data, which may not be what they intended. Some of what we add here may be reusable, some not, and some perhaps would be useful if other users could plug in their own warnings and we create a useful way to do that.

**Reply by cccs-tom** [reply]
I agree it would be great to have a proper Trino validator!
